app/models/team.rb

```diff
@@ -0,0 +1,38 @@
+class Team < ApplicationRecord
+  has_many :user_teams
+  has_many :members, through: :user_teams, source: :user, class_name: User
+
+  scope :related_to, -> (user_ids) do
+    joins(:user_teams)
+      .where(user_teams: { user_id: user_ids })
+      .distinct
+  end
+
+  validates_associated :user_teams
+
+  validates :name, presence: true, uniqueness: true
+
+  validate :unique_member_collections_for_teams
+
+  def member_ids
```

Comment:
yes but `unique_member_collections_for_teams` uses map directly not members_id so remove this method